### DISCUSSION TITLE
Size of preview and design window

### DISCUSSION THREAD

**Discussion by trixtan** [eclipse/windowbuilder#786] [Q&A]
I use default java templates with predefined components, one of which sets a default value for its bounds with `setBounds(5, 5, 600, 500);`
These bounds are bigger than the default size of the designer editor, and i get the following result:
![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/6442567/a49083d6-634d-4cd3-ae49-28806a958296)
Is there a way to programmatically set the size of the design ui when the java compilation unit is parsed?

**Comment by ptziegler** [comment]
WindowBuilder takes the bounds directly from the widgets. I don't think it makes much sense to change the bounds programmatically, as then the preview in the editor wouldn't match the actual widget.

Can you perhaps provide the source code so that I can have a closer look at what exactly is going on here?

**Reply by trixtan** [reply]
Hello,

thank you for your reply.
I do not know the reason, but since I change the Type of the parent composite to a custom one, everything works.
Now when I open the editor, everything looks good:
![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/6442567/1772a9fa-51b6-4d07-8169-0f379153c69f)
This is the code:
```import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Composite;

import com.ifx.davex.ui.controls.app.AppComposite;
import com.ifx.davex.ui.controls.app.GTab;
import com.ifx.davex.ui.controls.app.Tab;
import com.ifx.davex.ui.controls.app.TabFolder;

public class PPP_GUI_COMPOSITE extends AppComposite {

	public PPP_GUI_COMPOSITE(Composite parent, int style) {
		super(parent, style);
		
		TabFolder tabFolder = new TabFolder(this, SWT.NONE);
		tabFolder.setBounds(5, 5, 600, 500);
		
		GTab gTab = new GTab(tabFolder, SWT.NONE);
		gTab.setManifestObj(true);
		gTab.setEnabled(true);
		gTab.setVisible(true);
		gTab.setText("New Tab");
		
		Tab tab = new Tab(tabFolder, SWT.NONE);
		gTab.setControl(tab);

	}

	@Override
	protected void checkSubclass() {
		// Disable the check that prevents subclassing of SWT components
	}
}```

and this is the parent:
`package com.ifx.davex.ui.controls.app;

import org.eclipse.swt.layout.FillLayout;
import org.eclipse.swt.widgets.Composite;

public class AppComposite extends Composite {

    public AppComposite(Composite parent, int style) {
        super(parent, style);
        setLayout(new FillLayout());
    }

}`

I think, the deciding part is the `setLayout(new FillLayout());`, which was missing before
